cdap-archetypes/cdap-spark-scala-archetype/src/main/resources/archetype-resources/pom.xml

```diff
@@ -91,9 +83,38 @@
     <pluginManagement>
       <plugins>
         <plugin>
-          <groupId>org.scala-tools</groupId>
-          <artifactId>maven-scala-plugin</artifactId>
-          <version>2.15.2</version>
+          <groupId>net.alchim31.maven</groupId>
+          <artifactId>scala-maven-plugin</artifactId>
+          <version>3.2.2</version>
+          <configuration>
+            <args>
+              <arg>-target:jvm-1.7</arg>
+            </args>
+          </configuration>
+          <executions>
+            <execution>
+              <id>scala-add-source</id>
+              <goals>
+                <goal>add-source</goal>

```

Comment:
By default it happens at initialize phase.. http://davidb.github.io/scala-maven-plugin/add-source-mojo.html